src/main/java/ua/rent/masters/easystay/service/impl/BookingServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+    }
+
+    @Override
+    public List<BookingResponseDto> getByUserIdOrStatus(Long userId, BookingStatus bookingStatus) {
+        List<Booking> bookings;
+        if (userId != null && bookingStatus != null) {
+            bookings = bookingRepository.findByUserIdAndStatus(userId, bookingStatus)
+                    .orElseThrow(() -> new BookingException(
+                            "Can't find bookings with provided userId and status"));
+        } else if (userId != null) {
+            bookings = bookingRepository.findByUserId(userId)
+                    .orElseThrow(() -> new BookingException(
+                            "Can't find bookings with provided userId"));
+        } else if (bookingStatus != null) {
+            bookings = bookingRepository.findByStatus(bookingStatus)
+                    .orElseThrow(() -> new BookingException(
+                            "Can't find bookings with provided status"));
+        } else {
+            throw new BookingException(
+                    "Please provide at least one search parameter (userId or status)");
+        }
+
+        return bookings.stream()
+                .map(bookingMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<BookingResponseDto> getAll() {
+        return bookingRepository.findAll().stream()
+                .map(bookingMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookingResponseDto getById(Long bookingId) {
+        Booking booking = getBookingByIdOrThrowException(bookingId);
+        return bookingMapper.toDto(booking);
+    }
+
+    @Override
+    public BookingResponseDto updateById(Long bookingId,
+                                                BookingRequestUpdateDto requestUpdateDto) {
+        Booking booking = getBookingByIdOrThrowException(bookingId);
+        if (booking.getStatus() != BookingStatus.PENDING) {
+            throw new BookingException("You can update your booking"
+                    + " only with status PENDING");
+        }
+
+        if (requestUpdateDto.checkInDate() != null) {
+            booking.setCheckInDate(requestUpdateDto.checkInDate());
+        }
+        if (requestUpdateDto.checkOutDate() != null) {
+            booking.setCheckOutDate(requestUpdateDto.checkOutDate());
+        }
+        if (requestUpdateDto.accommodationId() != null) {
+            booking.setAccommodationId(requestUpdateDto.accommodationId());
+        }
+
+        bookingRepository.save(booking);
+
+        return bookingMapper.toUpdatedDto(booking);
+    }
+
+    @Override
+    public void deleteById(Long bookingId) {
+        Booking booking = getBookingByIdOrThrowException(bookingId);
+        bookingRepository.deleteById(booking.getId());
+    }
+
+    private void validateAccommodationId(Long accommodationId) {
+        if (accommodationId == null || accommodationId < 1) {
+            throw new AccommodationException("Please add accommodation");
+        }
+    }
```

Comment:
It might not be the case. If for example It was accomodation with id 7 but later it was deleted it will pass validation but it will be error. Best way is to select from accommodation service and check if you received some or not.